big_tests/tests/mod_event_pusher_rabbit_SUITE.erl

```diff
@@ -0,0 +1,321 @@
+    QueueBindings = presence_bindings(Queue, JIDs),
+    wait_for_exchanges_to_be_created(Connection, JIDs),
+    bind_queues_to_exchanges(Channel, QueueBindings),
+    subscribe_to_rabbit_queue(Channel, Queue).
+
+-spec wait_for_exchanges_to_be_created(Connection :: pid(),
+                                       Exchanges :: [binary()]) -> pid().
+wait_for_exchanges_to_be_created(Connection, Exchanges) ->
+    {ok, TestChannel} = amqp_connection:open_channel(Connection),
+    [is_exchange_present(Connection, TestChannel, Exchange,
+                         ?IF_EXCHANGE_EXISTS_RETRIES)
+     || Exchange <- Exchanges],
+    ok.
+
+-spec declare_rabbit_queue(Channel :: pid()) -> binary().
+declare_rabbit_queue(Channel) ->
+    #'queue.declare_ok'{queue = Queue} =
+        amqp_channel:call(Channel, #'queue.declare'{exclusive = true}),
+    Queue.
+
+-spec presence_bindings(Queue :: binary(), JIDs :: [binary()]) ->
+    [rabbit_binding()].
+presence_bindings(Queue, JIDs) ->
+    [{Queue, JID, presence_rk(JID)} || JID <- JIDs].
+
+-spec bind_queues_to_exchanges(Channel :: pid(),
+                               Bindings :: [rabbit_binding()]) ->
+    [amqp_client:amqp_method() | ok | blocked | closing].
+bind_queues_to_exchanges(Channel, Bindings) ->
+    [bind_queue_to_exchange(Channel, Binding) || Binding <- Bindings].
+
+bind_queue_to_exchange(Channel, {Queue, Exchange, RoutingKey}) ->
+    amqp_channel:call(Channel, #'queue.bind'{exchange = Exchange,
+                                             routing_key = RoutingKey,
+                                             queue = Queue}).
+
+-spec is_exchange_present(Connection :: pid(), Channel :: pid(),
+                                      Exchange :: binary(),
+                                      Retries :: non_neg_integer()) ->
+    true | no_return().
+is_exchange_present(_Connection, Channel, Exchange, 0) ->
+    amqp_channel:call(Channel, #'exchange.declare'{exchange = Exchange,
+                                                   type = <<"topic">>,
+                                                   passive = true}),
+    true;
+is_exchange_present(Connection, Channel, Exchange, Retries) ->
+    try amqp_channel:call(Channel, #'exchange.declare'{exchange = Exchange,
+                                                       type = <<"topic">>,
+                                                       passive = true}) of
+        {'exchange.declare_ok'} -> true
+    catch
+        _Error:_Reason ->
+            timer:sleep(?WAIT_FOR_EXCHANGE_INTERVAL),
+            %% The old channel `Channel` is closed.
+            {ok, NewChannel} = amqp_connection:open_channel(Connection),
+            is_exchange_present(Connection, NewChannel, Exchange, Retries - 1)
+    end.
+
+error_when_exchange_is_absent(Connection, Channel, Exchange, 0) ->
+    is_exchange_present(Connection, Channel, Exchange, 0);
+error_when_exchange_is_absent(Connection, Channel, Exchange, Retries) ->
```

Comment:
- You have double countdown - each call `error_when_exchange_is_absent(Connection, Channel, Exchange, Retries)` causes `Retries` calls to `is_exchange_present/4`
- I would rewrite this function. Body may be copied from `is_exchange_present/4`, but it will return true when exception is raised and retry when exchange is present. Code duplication is not significant, but it will look better in test case (asserting on boolean is better than expecting an exception).
- Rename to `is_exchange_absent/4`
- Be aware of countdown to `0` vs `1` as in `is_exchange_absent/4`
- Missing spec